### PR TITLE
Update CLI to use ISO_8601 time format.

### LINKED ISSUE #3806: Change the date format to a universal one, ISO_8601
### Nomad version
Nomad v0.7.0

### Operating system and Environment details
debian stretch

### Issue
Date format is US format only, that is very confusing for everyone else and may cause bad decisions.
Date should be using locale, or even better, ISO_8601. This format causes no confusion, alllow easy sorting and everyone will understand it
https://en.wikipedia.org/wiki/ISO_8601

### Reproduction steps
```
$ nomad job history -p e | grep Date

Submit Date = 09/12/17 15:25:33 UTC
```

Is this `2017/09/12` or `2017/12/09` ? or even `2009/12/17?`
ISO would be `2017-09-12T15:25:33Z`

This makes everything clear for everyone, not just US people

### PR DESCRIPTION
**Pull request by jrasell**
This change updates the formatTime CLI helper function to return an ISO_8601 time format which will make CLI time usage more consistent and easier. Previously the time format was in US style format which was somewhat confusing to non US users.

This has been tested locally:

```
./jrasell-nomad job history example
Version     = 0
Stable      = true
Submit Date = 2018-01-30T08:32:29Z
```

```
./jrasell-nomad alloc-status f1bab94a |grep 'Started At' 
Started At     = 2018-01-30T08:32:30Z
```

Closes #3806

### CODE REVIEW

**Comment by jippi** [comment]
looks good! my personal preference is RFC 2822 formatted dates, less dense to grok as a human :)

**Comment by danielmotaleite** [comment]
@jippi true, but they are also hard to parse (and sort) by other tools. ISO_8601 is a little harder to read by human, but very consistent and flexible

**Comment by danielmotaleite** [comment]
and thanks for the PR @jrasell

**Comment by chelseakomlo** [comment]
Thanks for this PR, looks good!
